src/renderer/screens/settings.tsx

```diff
@@ -0,0 +1,157 @@
+  import Container from 'renderer/components/container';
+  import { adjustRam } from 'renderer/services/ram.service';
+  import { adjustNick } from 'renderer/services/nickname.service';
+
+  import playerData from '../../common/player.json';
+  import ramData from '../../common/ram.json';
+  // Placeholder values for RAM adjustment
+  var plRamMin = '';
+  var plRamMax = '';
+  
+  interface StateNick {
+    value: string;
+    setValue: React.Dispatch<React.SetStateAction<string>>;
+  }
+  interface StateRam {
+    minval: string;
+    maxval: string;
+    setRammn: React.Dispatch<React.SetStateAction<string>>;
+    setRammx: React.Dispatch<React.SetStateAction<string>>;
+    selects: number;
+    setSelects: React.Dispatch<React.SetStateAction<number>>;
+  }
+  
+  // Valid chars in nickname
+  const VALID_NAME_CHARS = /^[a-zA-Z0-9-_]+$/;
+
+  // Check if nickname is valid and apply the changes
+  function checkNick(nick: string) {
+    for (let i = 0; i < nick.length; i += 1) {
+      if (!nick.match(VALID_NAME_CHARS) || nick.length < 3 ||  nick.length > 16) {
+        return true;
+      }
+      else {
+        // Adjust nickname
+        adjustNick(nick);
+      }
+    }
+  }
+
+  // Check if RAM value is valid and apply the changes
+  function checkRam(minimum: string, maximum: string, ramSize: number) {
+    var minvalue = parseInt(minimum);
+    var maxvalue = parseInt(maximum);
+
+    // If RAM type is GB
+    if(ramSize === 1) {
+      plRamMin = (parseInt(ramData.min) / 1024).toString();
+      plRamMax = (parseInt(ramData.max) / 1024).toString();
+
+      if(minimum == '' && maximum == '') return;
+      if(isNaN(minvalue) || isNaN(maxvalue)) return true;
+      if(minvalue > maxvalue) return true;
+      else if(minvalue < 1) return true;
+      else if(maxvalue > 32) return true;
+      adjustRam(minimum, maximum, ramSize);
+    }
+    
+    // If RAM type is MB
+    if(ramSize === 0) {
+      plRamMin = ramData.min;
+      plRamMax = ramData.max;
+
+      if(minimum == '' && maximum == '') return;
+      if(isNaN(minvalue) || isNaN(maxvalue)) return true;
+      if (minvalue < 1024) return true;
+      else if (minvalue > maxvalue) return true;
+      else if(maxvalue > 32768) return true;
+      adjustRam(minimum, maximum, ramSize);
+    }
+  };
+  
+  // Nickname input
+  function Nickname({ value, setValue }: StateNick) {
+    const invalidName = checkNick(value);
+  
+    return (
+      <Flex display={"column"} width="60%" alignContent={"center"}>
+        <Box width={"100%"}>
+          {invalidName}
+        </Box>
+
+        <FormLabel>Nickname:</FormLabel>
+        <Input
+          placeholder={playerData.nick}
+          variant="flushed"
+          value={value}
+          onChange={(nick) => setValue(nick.target.value)}
+          isInvalid={invalidName}
+        />
+      </Flex>
+    );
+  }
+
+  // RAM input
+  function Ram({ minval, maxval, setRammn, setRammx, selects, setSelects }: StateRam) {
```

Comment:
It is bad practice to shorten variable names.

It is preferable to put "minRam" or "minValu"e instead of "minval"